src/elementPlugins/shortText/src/ApplicationView.tsx

```diff
@@ -7,52 +7,25 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
   onUpdate,
   initialValue,
   isEditable,
-  isRequired,
-  allResponses,
-  evaluator,
+  currentResponse,
+  validationState,
+  onSave,
 }) => {
-  const [validationMessageDisplay, setValidationMessageDisplay] = useState('')
   const [value, setValue] = useState(initialValue?.text)
-  const [isValid, setIsValid] = useState<boolean>()
-  const [responses, setResponses] = useState({
-    thisResponse: undefined,
-    ...allResponses,
-  })
-  const {
-    validation: validationExpression,
-    validationMessage,
-    placeholder,
-    maskedInput,
-    label,
-  } = templateElement?.parameters
+  const { placeholder, maskedInput, label } = templateElement?.parameters
 
   useEffect(() => {
-    // Do validation, setIsValid
-    if (validationExpression && responses.thisResponse !== undefined) {
-      evaluator(validationExpression, { objects: [responses], APIfetch: fetch })
-        .then((result: boolean) => {
-          setIsValid(result)
-        })
-        .catch((err: any) => console.log(err))
-    } else setIsValid(true)
-  }, [responses])
-
-  useEffect(() => {
-    setResponses({ thisResponse: value, ...allResponses })
-  }, [value])
-
-  useEffect(() => {
-    if (isValid) setValidationMessageDisplay('')
-    else setValidationMessageDisplay(validationMessage)
-  }, [isValid])
+    onUpdate(value)
+  }, [])
 
   function handleChange(e: any) {
+    onUpdate(e.target.value)
     setValue(e.target.value)
   }
 
   function handleLoseFocus(e: any) {
     // TO-DO if password (i.e 'maskedInput'), do HASH on password before sending value
-    onUpdate({ value: { text: value }, isValid })
+    onSave({ text: value })
   }
 
   return (
```

Comment:
This one is `Form.input`, should we simply change it to `Input` for now, since its form would be in an external component?

Reply:
Form.input and Input have slightly different properties, and this is what Andrei configured originally. I tried Input, and it wasn't as appropriate for some reason (though I can't now remember what the reason was). This will be the kind of thing we establish in our ongoing style/layout discussion.

Reply:
@nmadruga i actually used Form.input because I saw some one using it for the original form (I think it was you). it might give us more control over layout